tools/@aws-cdk/prlint/lint.ts

```diff
@@ -272,11 +304,92 @@ export class PullRequestLinter {
     }
   }
 
+  /**
+   * Whether or not the codebuild job for the given commit is successful
+   *
+   * @param sha the commit sha to evaluate
+   */
+  private async codeBuildJobSucceeded(sha: string): Promise<boolean> {
+    const statuses = await this.client.rest.repos.listCommitStatusesForRef({
+      owner: this.prParams.owner,
+      repo: this.prParams.repo,
+      ref: sha,
+    });
+    return statuses.data.some(status => status.context === CODE_BUILD_CONTEXT && status.state === 'success');
+  }
+
+  public async validateStatusEvent(pr: GitHubPr, status: StatusEvent): Promise<void> {
+    if (status.context === CODE_BUILD_CONTEXT && status.state === 'success') {
+      await this.assessNeedsReview(pr);
+    }
+  }
+
+  /**
+   * Assess whether or not a PR is ready for review from a core team member.
+   * This is needed because some things that we need to evaluate are not filterable on
+   * the builtin issue search. A PR is ready for review when:
+   *
+   *   1. Not a draft
+   *   2. Does not have any merge conflicts
+   *   3. PR linter is not failing OR the user has requested an exemption
+   *   4. A maintainer has not requested changes
+   */
+  private async assessNeedsReview(
+    pr: Pick<GitHubPr, "mergeable_state" | "draft" | "labels" | "number">,
+  ): Promise<void> {
+    // only care about the codebuild build job
+    // the other info we will get from the PR itself
+
+    const reviews = await this.client.pulls.listReviews(this.prParams);
+    const maintainerRequestedChanges = reviews.data.some(review => review.author_association === 'MEMBER' && review.state === 'CHANGES_REQUESTED');
```

Comment:
can you confirm that the automated PR review is not a part of `author_association === 'MEMBER'`?

Reply:
Good catch. I'll check on that.

Reply:
Ok the automated PR review is from `aws-cdk-automation` and they are a `COLLABORATOR`.

Reply:
How does github differentiate between us and `aws-cdk-automation` then?

Reply:
It looks like author_association is not about permissions (aws-cdk-automation and our team are both admins).

MEMBER = "Author is a member of the organization that owns the repository."

COLLABORATOR = "Author has been invited to collaborate on the repository."

I'll add a code comment.

Reply:
Ok added a comment.

Reply:
Ok. `aws-cdk-automation` is a weird case still. It's invited as a collaborator on the repository, but it's also a member of the org that owns the repository (aws), for now.

I'll take your word for it that it's listed as a `COLLABORATOR`, but pointing out that `MEMBER` and `COLLABORATOR` may not be mutex.